### PR TITLE
Remove shell integration plugin for GNOME's Nautilus

### LINKED ISSUE #11150: Remove nautilus code that was moved to https://github.com/owncloud/client-desktop-shell-integration-nautilus

### PR DESCRIPTION
**Pull request by fmoc**
... and its derivatives.

Moved to https://github.com/owncloud/client-desktop-shell-integration-nautilus.

Fixes #11150.

### CODE REVIEW

**Review by TheOneRing — APPROVED** [review]

**Comment by fmoc** [comment]
Looks like the GUI tests need some update to their code. @saw-jan could you have a look, please?

File: test/gui/shared/scripts/helpers/SyncHelper.py [hunk 1]
```diff
@@ -1,17 +1,41 @@
...
+from helpers.FilesHelper import sanitizePath
+from helpers.ConfigHelper import get_config
+
+# NOTE: 'syncstate.py' was removed from client
+# and is now available at 'owncloud/client-desktop-shell-integration-nautilus'
+# check if 'syncstate.py' is available, if not, download it
+custom_lib = get_config('custom_lib')
+syncstate_lib_file = os.path.join(custom_lib, 'syncstate.py')
+if not os.path.exists(custom_lib):
+    os.makedirs(get_config('custom_lib'), exist_ok=True)
```
Comment (fmoc) [inline test/gui/shared/scripts/helpers/SyncHelper.py]:
```suggestion
os.makedirs(get_config('custom_lib'), exist_ok=True)
```
Check is not needed, you use `exist_ok=True` anyway.

File: test/gui/shared/scripts/helpers/SyncHelper.py [hunk 2]
```diff
@@ -1,17 +1,41 @@
...
-from os.path import realpath
+import os
+import urllib
 from squish import waitFor, snooze
 
+from helpers.FilesHelper import sanitizePath
+from helpers.ConfigHelper import get_config
+
+# NOTE: 'syncstate.py' was removed from client
+# and is now available at 'owncloud/client-desktop-shell-integration-nautilus'
```
Comment (fmoc) [inline test/gui/shared/scripts/helpers/SyncHelper.py:12]:
```suggestion
# and is now available at https://github.com/owncloud/client-desktop-shell-integration-nautilus
```

File: test/gui/shared/scripts/helpers/SyncHelper.py [hunk 3]
```diff
@@ -1,17 +1,41 @@
...
+syncstate_lib_file = os.path.join(custom_lib, 'syncstate.py')
+if not os.path.exists(custom_lib):
+    os.makedirs(get_config('custom_lib'), exist_ok=True)
+if not os.path.exists(syncstate_lib_file):
+    URL = "https://raw.github.com/owncloud/client-desktop-shell-integration-nautilus/master/src/syncstate.py"
+    try:
+        urllib.request.urlretrieve(
+            URL, os.path.join(get_config('custom_lib'), 'syncstate.py')
+        )
+    except urllib.error.HTTPError as e:
```
Comment (fmoc) [inline test/gui/shared/scripts/helpers/SyncHelper.py:22]: I think you can just have it raise the exception, we are all developers and can read it. Also, there is not much of a point in "rewriting" the exception to `Exception` either.

File: test/gui/shared/scripts/helpers/SyncHelper.py [hunk 4]
```diff
@@ -1,17 +1,39 @@
...
+
+# NOTE: 'syncstate.py' was removed from client
+# and is now available at 'owncloud/client-desktop-shell-integration-nautilus'
+# check if 'syncstate.py' is available, if not, download it
+custom_lib = get_config('custom_lib')
+syncstate_lib_file = os.path.join(custom_lib, 'syncstate.py')
+if not os.path.exists(custom_lib):
+    os.makedirs(custom_lib, exist_ok=True)
+if not os.path.exists(syncstate_lib_file):
+    URL = "https://raw.github.com/owncloud/client-desktop-shell-integration-nautilus/master/src/syncstate.py"
```
Comment (fmoc) [inline test/gui/shared/scripts/helpers/SyncHelper.py:19]: Please use a lower-case variable name like `url`. Using uppercase only for variable names violates Python conventions.

**Pull request merged**